src/commands/general/ping.rs

```diff
@@ -1,11 +1,34 @@
+use std::time::{Duration, Instant};
+
 use crate::{Context, Error};
 
 use log::trace;
+use poise::CreateReply;
+
+const PING_PREFIX: &str = "<:catstareback:1078622789885497414> Pong!";
 
 /// Replies with pong!
 #[poise::command(slash_command, prefix_command, track_edits = true, ephemeral)]
 pub async fn ping(ctx: Context<'_>) -> Result<(), Error> {
 	trace!("Running ping command!");
-	ctx.say("Pong!").await?;
+
+	let start = Instant::now();
+	let response = ctx.say(PING_PREFIX).await?;
+
+	let rtt = start.elapsed().as_millis();
+	let gateway_ping = match ctx.ping().await {
+		Duration::ZERO => "Undetermined".to_string(),
+		duration => format!("{}ms", duration.as_millis()),
+	};
+
+	response
+		.edit(
+			ctx,
+			CreateReply::default().content(format!(
+				"{PING_PREFIX}\n\nRTT: {rtt}ms\nGateway: {gateway_ping}",
+			)),
+		)
+		.await?;
+
```

Comment:
this is how it was implemented before. i don't see many cases where this is extremely useful in the first place, let alone RTT as well; so i don't see a reason to "improve" on the previous implementation either
```suggestion
  let resp = format!("Pong! (`{}ms`)", ctx.ping().await.as_millis());
  ctx.say(resp).await?;
```